core/src/main/java/org/mskcc/cbio/oncokb/util/VariantConsequenceUtils.java

```diff
@@ -94,6 +95,46 @@ private static void cacheVariantConsequencesMap() {
         }
     }
 
+    public static VariantConsequence consequenceResolver(String consequenceTerms) {
+        return consequenceResolver(consequenceTerms, null);
+    }
+
+    // The goal is to reserve the details of the consequence if possible, but when consequence conflicts with variant class, we need to decide which one to use
+    public static VariantConsequence consequenceResolver(String consequenceTerms, String variantClass) {
+        if (StringUtils.isEmpty(consequenceTerms) && StringUtils.isEmpty(variantClass)) {
+            return null;
+        }
+        if (consequenceTerms == null) {
+            consequenceTerms = "";
+        }
+        if (variantClass == null) {
+            variantClass = "";
+        }
+
+        VariantConsequence consequence = findVariantConsequenceByTerm(consequenceTerms.contains(",") ? pickConsequenceTerm(consequenceTerms) : consequenceTerms);
+        VariantConsequence variantClassConsequence = variantClass == null ? null : findVariantConsequenceByTerm(variantClass);
+
+        if (consequence == null) {
+            if (variantClassConsequence != null) {
+                return variantClassConsequence;
+            } else {
+                return null;
+            }
+        } else {
+            if (variantClassConsequence != null && variantClassConsequence != consequence) {
```

Comment:
```suggestion
            if (variantClassConsequence != null && variantClassConsequence.equals(consequence)) {
```
Should we use the `equals` method that is defined in `VariantConsequence` class? `!=` will just be comparing the references to the two object.